File: 151051/Komar/task310-rest/src/main/java/by/bsuir/resttask/repository/Repository.java
package by.bsuir.resttask.repository;

import jakarta.annotation.Nonnull;

import java.util.List;
import java.util.Optional;

public interface Repository<T, ID> {
    void delete(@Nonnull T entity);

    void deleteById(ID id);

    List<T> findAll();

    Optional<T> findById(ID id);

    <S extends T> S save(@Nonnull S entity);
}

File: 251003/Pankratiev/rest/src/main/kotlin/bsuir/dc/rest/repository/Repository.kt
package bsuir.dc.rest.repository

import bsuir.dc.rest.entity.Identifiable

interface Repository<T : Identifiable> {
    fun findById(id: Long): T
    fun findAll(): List<T>
    fun save(entity: T): T
    fun update(entity: T): T
    fun deleteById(id: Long)
}
